### PR TITLE
Implement chirp w/ hashtag

### PR DESCRIPTION
**Pull request by apetranik**

### CODE REVIEW

File: service/test/servicelayerinstance.cc [hunk 1]
```diff
@@ -88,6 +88,36 @@ bool ServiceLayerInstance::Chirp(const std::string &username,
     kvstore->Put("reply" + parentid,
                  reply_chirp_string_value);  // Add reply<ID> to backend
   }
+  /* Check for hashtag in chirp - if so add to KVS as a hashtag chirp */
+  std::size_t start = text.find(" #");
+  if (start != -1) {
+    std::string hashtag = text.substr(start + 1);  // +1 gets rid of space
+    std::size_t end = hashtag.find(" ");           // cut off at end of hashtag
```
Comment (cindyciclta) [inline service/test/servicelayerinstance.cc]: comments should be using the same type. /* */

File: service/servicelayerfunctionalities.cc [hunk 2]
```diff
@@ -162,6 +162,40 @@ Status ServerForCommandLineClient::chirp(ServerContext *context,
...
     client_key.put("reply" + request->parent_id(), reply_chirp_to_proto);
   }
+
+  /* Check for hashtag in chirp - if so add to KVS as a hashtag chirp */
+  std::size_t start = request->text().find(" #");
+  if (start != -1) {
+    std::string hashtag =
+        request->text().substr(start + 1);  // +1 gets rid of space
+    std::size_t end = hashtag.find(" ");    // cut off at end of hashtag
+    hashtag = hashtag.substr(0, end);
```
Comment (cindyciclta) [inline service/servicelayerfunctionalities.cc]: Same here with the comments~

File: service/servicelayerfunctionalities.cc [hunk 3]
```diff
@@ -162,6 +162,40 @@ Status ServerForCommandLineClient::chirp(ServerContext *context,
     }
     client_key.put("reply" + request->parent_id(), reply_chirp_to_proto);
   }
+
+  /* Check for hashtag in chirp - if so add to KVS as a hashtag chirp */
```
Comment (cindyciclta) [inline service/servicelayerfunctionalities.cc]: Maybe put this part in another function since chirp() is pretty long already?

Comment (apetranik) [reply service/servicelayerfunctionalities.cc]: good idea

File: service/test/servicelayerinstance.h [hunk 4]
```diff
@@ -44,10 +44,15 @@ class ServiceLayerInstance {
   */
   chirp::Chirp ConvertToChirp(std::string byte);
   /*
+  Converts serialized string to chirp::Hashtag
+  */
+  std::vector<chirp::Chirp> ConvertToHashtag(std::string byte);
```
Comment (barath) [inline service/test/servicelayerinstance.h]: Const reference?

File: service/test/servicelayerinstance.h [hunk 5]
```diff
@@ -44,10 +44,15 @@ class ServiceLayerInstance {
   */
   chirp::Chirp ConvertToChirp(std::string byte);
   /*
+  Converts serialized string to chirp::Hashtag
```
Comment (barath) [inline service/test/servicelayerinstance.h]: Comment doesn't seem to match method signature.

File: service/test/servicelayerinstance.cc [hunk 6]
```diff
@@ -88,6 +88,36 @@ bool ServiceLayerInstance::Chirp(const std::string &username,
     kvstore->Put("reply" + parentid,
                  reply_chirp_string_value);  // Add reply<ID> to backend
   }
+  /* Check for hashtag in chirp - if so add to KVS as a hashtag chirp */
+  std::size_t start = text.find(" #");
```
Comment (barath) [inline service/test/servicelayerinstance.cc]: Move string parsing to other method?

Comment (apetranik) [reply service/test/servicelayerinstance.cc]: @barath responded to all PR comments

File: service/servicelayerfunctionalities.cc [hunk 7]
```diff
@@ -439,3 +410,42 @@ bool ServerForCommandLineClient::CheckIfReplyIDExist(
...
+    if (from_get_function.size() != 0) {
+      LOG(INFO) << "Hashtag has been used before" << std::endl;
+      std::string getValue = from_get_function[0];
+      tag.ParseFromString(getValue);
+    }
+
+    // Add chirp to Hashtag
+    chirp::Chirp *new_message;
+    std::string hashtag_chirp;
+    {
```
Comment (barath) [inline service/servicelayerfunctionalities.cc:438]: Why the new scope?  I don't see any vars that might go out of scope at the end.

File: service/servicelayerfunctionalities.cc [hunk 8]
```diff
@@ -439,3 +410,42 @@ bool ServerForCommandLineClient::CheckIfReplyIDExist(
...
+
+void ServerForCommandLineClient::HandleChirpHashTag(
+    const std::string &username, const std::string &text,
+    const std::string &parent_id, const std::string &next_chirp_ID,
+    std::time_t seconds, int64_t microseconds_since_epoch,
+    ClientForKeyValueStore &client_key) {
+  /* Check for hashtag in chirp - if so add to KVS as a hashtag chirp */
+  std::size_t start = text.find(" #");
+  if (start != -1) {
+    std::string hashtag = text.substr(start + 1); /* +1 gets rid of space */
```
Comment (barath) [inline service/servicelayerfunctionalities.cc:422]: Is it possible to move parsing to a separate function?

**Pull request merged**